pkg/infrastructure/clusterapi/clusterapi.go

```diff
@@ -120,6 +120,7 @@ func (i *InfraProvider) Provision(dir string, parents asset.Parents) ([]*asset.F
 	for _, m := range infraManifests {
 		m.SetNamespace(capiutils.Namespace)
 		if err := cl.Create(ctx, m); err != nil {
+			clusterapi.System().Teardown()
```

Comment:
```suggestion
```

Reply:
Change Provision() to look like:

```go
func (i *InfraProvider) Provision(dir string, parents asset.Parents) (_ []*asset.File, retErr error) {
```

then add a defer like:
```go
defer func() {
    if retErr != nil {
        clusterapi.System().Teardown()
    }
}()
```

Reply:
If the clusterapi system "leaks" from the `Provision()` function to be shutdown at a later time, couldn't the caller of `Provision` do the shutdown in case of error? The `defer` function is still feeble since it's very easy to miss setting the `retErr` variable before returning in case of error.

Reply:
Yeah that can work, but I was under the impression that given this is an asset, we might want to do within `Provision` here. I'm fine either way, the suggestion above is mostly a pattern I've seen elsewhere work well